internal/resources/app_endpoint_oidc_default.go

```diff
@@ -0,0 +1,274 @@
+	var plan providerschema.AppEndpointDefaultOidcProvider
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := plan.OrganizationId.ValueString()
+	projectId := plan.ProjectId.ValueString()
+	clusterId := plan.ClusterId.ValueString()
+	appServiceId := plan.AppServiceId.ValueString()
+	appEndpointName := plan.AppEndpointName.ValueString()
+	providerId := plan.ProviderId.ValueString()
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/oidcProviders/defaultProvider",
+		r.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		appEndpointName,
+	)
+
+	payload := api.AppEndpointOIDCDefaultProviderRequest{ProviderID: providerId}
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusNoContent}
+	_, err := r.Client.ExecuteWithRetry(ctx, cfg, payload, r.Token, map[string]string{"Content-Type": "application/json"})
+	if err != nil {
+		resp.Diagnostics.AddError("Error Setting Default OIDC Provider", api.ParseError(err))
+		return
+	}
+
+	// Refresh state by listing providers and picking the default
+	selected, err := r.waitForDefaultProvider(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName, providerId)
+	if err != nil {
+		resp.Diagnostics.AddWarning("Error reading default OIDC provider after create", api.ParseError(err))
+	} else if selected.ProviderID != "" {
+		plan.ProviderId = types.StringValue(selected.ProviderID)
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+// Read finds the current default provider by listing available providers.
+func (r *AppEndpointDefaultOidcProvider) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state providerschema.AppEndpointDefaultOidcProvider
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError("Error Reading Default OIDC Provider", "Could not validate state: "+err.Error())
+		return
+	}
+
+	selected, err := r.getDefaultProvider(ctx, IDs[providerschema.OrganizationId], IDs[providerschema.ProjectId], IDs[providerschema.ClusterId], IDs[providerschema.AppServiceId], IDs[providerschema.AppEndpointName])
+	if err != nil {
+		// If list returns 404, remove from state
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Error Reading Default OIDC Provider", "Could not list OIDC providers: "+errString)
+		return
+	}
+
+	if selected.ProviderID == "" {
+		// No default currently set; remove from state so plan can set it
+		tflog.Info(ctx, "No default OIDC provider found; removing from state")
+		resp.State.RemoveResource(ctx)
+		return
+	}
```

Comment:
no need for this check

`getDefaultProvider` should return error if for some reason there is no default in non-empty list of oidc providers

Reply:
https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/396/commits/1d43abad6f6f7afa0d10c420eb59e31852c3f3eb